### PR TITLE
Rcal 624 regtest updates for B11, rad & roman_datamodels pin

### LINKED ISSUE #815: Update regression tests for 23Q4_B11
_Issue [RCAL-624](https://jira.stsci.edu/browse/RCAL-624) was created on JIRA by [David Davis](https://jira.stsci.edu/secure/ViewProfile.jspa?name=ddavis):_

RAD and roman_datamodels have updates to the L1 models, once we get new input data files the input files on artifactory will need to be updated as well as the truth files.

### PR DESCRIPTION
**Pull request by ddavis-stsci**
Resolves [RCAL-624](https://jira.stsci.edu/browse/rcal-624)

Closes #815 

This PR updates the artifactory files & updates some of the input file names for the regtest for refpix and flatfield tests.
The passing regression tests are at,
https://plwishmaster.stsci.edu:8081/job/RT/job/Roman-Developers-Pull-Requests/317/
**Checklist**
- [N/A] added entry in `CHANGES.rst` under the corresponding subsection
- [x] updated relevant tests
- [n/a] updated relevant documentation
- [x] updated relevant milestone(s)
- [x] added relevant label(s)

### CODE REVIEW

File: requirements-dev.txt [hunk 1]
```diff
@@ -6,8 +6,8 @@ git+https://github.com/astropy/asdf-astropy
 git+https://github.com/spacetelescope/crds
 git+https://github.com/spacetelescope/gwcs
 numpy>=0.0.dev0
-git+https://github.com/spacetelescope/roman_datamodels@0.17.0
-git+https://github.com/spacetelescope/rad@0.17.0
+git+https://github.com/spacetelescope/roman_datamodels@0.17.1
+git+https://github.com/spacetelescope/rad@0.17.1
```
Comment (WilliamJamieson) [inline requirements-dev.txt:10]:
```suggestion
git+https://github.com/spacetelescope/roman_datamodels
git+https://github.com/spacetelescope/rad
```
These should not be pinned to a specific versions. The devdeps tests are supposed to test against the development versions of packages not some specific release.

**Review by WilliamJamieson — CHANGES_REQUESTED** [review]

File: pyproject.toml [hunk 2]
```diff
@@ -14,18 +14,18 @@ classifiers = [
...
+    'astropy >=5.3.0',
     'crds >=11.16.16',
     'gwcs >=0.18.1',
     'jsonschema >=4.8',
     'numpy >=1.22',
     'photutils >=1.6.0',
     'pyparsing >=2.4.7',
     'requests >=2.22',
-    'rad == 0.17.0',
+    'rad == 0.17.1',
```
Comment (WilliamJamieson) [inline pyproject.toml:25]:
```suggestion
    'rad >=0.17.1',
```
This should not be exact pinned.

File: pyproject.toml [hunk 3]
```diff
@@ -14,18 +14,18 @@ classifiers = [
...
 #    'rad >= 0.16.0, < 0.17.1',
 #    'rad @ git+https://github.com/spacetelescope/rad.git@main',
-    'roman_datamodels ==0.17.0',
+    'roman_datamodels ==0.17.1',
```
Comment (WilliamJamieson) [inline pyproject.toml:28]:
```suggestion
    'roman_datamodels >=0.17.1',
```
This should not be exact pinned.

**Review by WilliamJamieson — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 3 above)
Comment (ddavis-stsci) [reply pyproject.toml:28]: Why, what purpose does that serve for a release?

Comment (WilliamJamieson) [reply pyproject.toml:28]: Exact pinning like this should be on a release branch not a development branch, if it needs to be done at all. Exact pins like these will interfere with development of `rad` and `roman_datamodels` by requiring us to make the changes I requested here in order to move forward with them. This is not a concern on a release branch, only on the development branch.

**Review by schlafly — APPROVED** [review]
This looks fine to me.

@WilliamJamieson , I hear your concerns re exact pinning.  But we're trying to get this out the door, and Dave & Zach plan to replace the exact pins with open ended versions immediately following the release.  So let's plan to talk about alternative release processes at an upcoming tag up (e.g., tomorrow), and let this go through as is.

**Comment by WilliamJamieson** [comment]
> @WilliamJamieson , I hear your concerns re exact pinning. But we're trying to get this out the door, and Dave & Zach plan to replace the exact pins with open ended versions immediately following the release. So let's plan to talk about alternative release processes at an upcoming tag up (e.g., tomorrow), and let this go through as is.

Regardless, the `requirements-dev.txt` should not be specifying a tag as it is.

**Review by WilliamJamieson — APPROVED** [review]

**Pull request merged**